viadot/sources/eurostat.py

```diff
@@ -0,0 +1,256 @@
+import pandas as pd
+
+from .base import Source
+from viadot.utils import handle_api_response
+
+
+class Eurostat(Source):
+    """
+    Class for creating instance of Eurostat connector to REST API by HTTPS response (no credentials required).
+    """
+
+    def __init__(
+        self,
+        dataset_code: str,
+        params: dict = None,
+        *args,
+        **kwargs,
+    ):
+        """
+        It is using HTTPS REST request to pull the data.
+        No API registration or API key are required.
+        Data will pull based on parameters provided in dynamic part of the url.
+
+        Example of url: https://ec.europa.eu/eurostat/api/dissemination/statistics/1.0/data/TEIBS020/?format=JSON&lang=EN&indic=BS-CSMCI-BAL
+        Static part: https://ec.europa.eu/eurostat/api/dissemination/statistics/1.0/data
+        Dynamic part: /TEIBS020/?format=JSON&lang=EN&indic=BS-CSMCI-BAL
+
+        Please note that for one dataset there are usually multiple data regarding different subjects.
+        In order to retrive data that you are interested in you have to provide parameters with codes into 'params'.
+
+        Args:
+            dataset_code (str): The code of eurostat dataset that we would like to upload - ALWAYS REQUIRED
+            params (dict, optional):
+                A dictionary with optional URL parameters. The key represents the parameter id, while the value is the code
+                for a specific parameter, for example: params = {'unit': 'EUR'} where "unit" is the parameter that you would like to set
+                and "EUR" is the code of the specific parameter. You can add more than one parameter, but only one code per parameter!
+                So you CAN NOT provide list of codes as in example 'params = {'unit': ['EUR', 'USD', 'PLN']}'
+                These parameters are REQUIRED in most cases to pull a specific dataset from the API.
+                Both parameter and code has to be provided as a string!
+                Defaults to None.
+        """
+
+        self.dataset_code = dataset_code
+        self.params = params
+        self.url = (
+            f"https://ec.europa.eu/eurostat/api/dissemination/statistics/1.0/data/"
```

Comment:
Yeah, it is actually good idea, because if something will change in this part of the url it will be easier to change this as environmental variable (it will automatically apply to all instances of the class).

Reply:
Created global variable BASE_URL - modyfication in source, task and flow